opentelemetry-instrumentation/src/opentelemetry/instrumentation/propagators.py

```diff
@@ -0,0 +1,121 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module implements experimental propagators to inject trace response context
+into HTTP responses. This is useful for server side frameworks that start traces
+when server requests and want to share the trace context with the client so the
+client can add it's spans to the same trace.
+
+This is part of an upcoming W3C spec and will eventually make it to the Otel spec.
+
+https://w3c.github.io/trace-context/#trace-context-http-response-headers-format
+"""
+
+import typing
+from abc import ABC, abstractmethod
+
+import opentelemetry.trace as trace
+from opentelemetry.context.context import Context
+from opentelemetry.propagators import textmap
+from opentelemetry.trace import format_span_id, format_trace_id
+
+_HTTP_HEADER_ACCESS_CONTROL_EXPOSE_HEADERS = "Access-Control-Expose-Headers"
+_BACK_PROPAGATOR = None
+
+
+def get_global_back_propagator():
+    return _BACK_PROPAGATOR
+
+
+def set_global_back_propagator(propagator):
+    global _BACK_PROPAGATOR  # pylint:disable=global-statement
+    _BACK_PROPAGATOR = propagator
+
+
+class DictHeaderSetter:
+    def set(self, carrier, key, value):  # pylint: disable=no-self-use
+        old_value = carrier.get(key, "")
+        if old_value:
+            value = "{0}, {1}".format(old_value, value)
+        carrier[key] = value
+
+
+default_setter = DictHeaderSetter()
+
+
+class FuncSetter:
+    def __init__(self, func):
+        self._func = func
+
+    def set(self, carrier, key, value):
+        self._func(carrier, key, value)
+
+
+class ResponsePropagator(ABC):
+    @abstractmethod
+    def _header_name(self) -> str:
+        pass
+
+    @abstractmethod
+    def _format_header(self, span_context) -> str:
+        pass
+
+    def inject(
+        self,
+        carrier: textmap.CarrierT,
+        context: typing.Optional[Context] = None,
+        setter: textmap.Setter = default_setter,
+    ) -> None:
+        """Injects SpanContext into the HTTP response carrier."""
+        span = trace.get_current_span(context)
+        span_context = span.get_span_context()
+        if span_context == trace.INVALID_SPAN_CONTEXT:
+            return
+
+        header_name = self._header_name()
+        setter.set(carrier, header_name, self._format_header(span_context))
+        setter.set(
+            carrier, _HTTP_HEADER_ACCESS_CONTROL_EXPOSE_HEADERS, header_name,
+        )
+
+
+class TraceResponsePropagator(ResponsePropagator):
```

Comment:
Would TraceContextPropagator make more sense? It is already deriving from the `ResponsePropagator` and `ServerTimingPropagator` does not have the word `Response` in it. I guess the only confusing thing would be if people were to mix it up with the regular propagator for tracecontext.

Reply:
How about we add "response" to both derived classes so we have `ServerTimingResponsePropagator` and `TraceResponsePropagator`?